### ISSUE TITLE
resize_images: Alignment of upsampled image seems incorrect with method=NEAREST_NEIGHBOR and align_corners=True

### ISSUE THREAD

**Issue by jvlmdr** [tensorflow/tensorflow#10989]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS Linux release 7.2.1511 (Core)
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: ('v1.1.0-0-g1ec6ed5', '1.1.0')
- **Bazel version (if compiling from source)**: 0.4.5- (@non-git)
- **CUDA/cuDNN version**: 7.5/5.1
- **GPU model and memory**: M40, 12GB
- **Exact command to reproduce**:

### Describe the problem

My understanding of the behaviour of `tf.image.resize_images` with `align_corners=True` is that it would resize
```
[A B C D] => [A x x B x x C x x D]
```
if the input size was 4 and the output size was 3*(4-1)+1=10, where the `x` values would be determined by the method of interpolation.
(I will provide 1D examples, although I am actually working with images.)

I see the expected behaviour with `method=tf.image.ResizeMethod.BILINEAR`:
```
[0 3 6 9] => [0 1 2 3 4 5 6 7 8 9]
```
however, with `method=tf.image.ResizeMethod.NEAREST_NEIGHBOR` the result is not what I expect:
```
[0 3 6 9] => [0 0 0 3 3 3 6 6 6 9]    # actual result
[0 3 6 9] => [0 0 3 3 3 6 6 6 9 9]    # desired result
```
Note that the nearest neighbor of the third `0` in the "actual result" is the element with value `3`.

### Source code / logs
I have attached a simple example that demonstrates this effect: [nearest.py.zip](https://github.com/tensorflow/tensorflow/files/1095366/nearest.py.zip)

**Comment by ppwwyyxx** [comment]
#6720
Seems to be fixed already.

**Comment by jvlmdr** [comment]
Oh thanks! It is fixed in 1.2.0
